extensions/cornerstone-dicom-rt/src/utils/promptHydrateRT.ts

```diff
@@ -0,0 +1,70 @@
+import hydrateRTDisplaySet from './_hydrateRT';
+
+const RESPONSE = {
+  NO_NEVER: -1,
+  CANCEL: 0,
+  HYDRATE_SEG: 5,
+};
+
+function promptHydrateRT({
+  servicesManager,
+  rtDisplaySet,
+  viewportIndex,
+  toolGroupId = 'default',
+}) {
+  const { UIViewportDialogService } = servicesManager.services;
+
+  return new Promise(async function(resolve, reject) {
+    const promptResult = await _askHydrate(
+      UIViewportDialogService,
+      viewportIndex
+    );
+
+    if (promptResult === RESPONSE.HYDRATE_SEG) {
+      const isHydrated = await hydrateRTDisplaySet({
+        rtDisplaySet,
+        viewportIndex,
+        toolGroupId,
+        servicesManager,
+      });
+
+      resolve(isHydrated);
+    }
+  });
+}
+
+function _askHydrate(UIViewportDialogService, viewportIndex) {
+  return new Promise(function(resolve, reject) {
+    const message = 'Do you want to open this Segmentation?';
```

Comment:
We should consider localizing the various strings for RT struct.

Reply:
it is hard since these are conditional prompts and we can not have hooks (useTranslation) inside conditional components. Can we do this later?

Reply:
yes this can be done later